go/src/lib/lomcommon/helper.go

```diff
@@ -0,0 +1,254 @@
+}
+
+
+func LogDebug(s string, a ...interface{})  {
+    LogMessage(syslog.LOG_DEBUG, s, a...)
+}
+
+var uuid_suffix = 0
+var UUID_BIN = "uuidgen"
+
+func GetUUID() string {
+    if newUUID, err := exec.Command(UUID_BIN).Output(); err != nil {
+        LogError("Internal failed uuidgen. (%s)", err)
+        uuid_suffix++
+        return fmt.Sprintf("%v-%d", time.Now().Unix(), uuid_suffix)
+    } else {
+        return string(newUUID)[:36]
+    }
+}
+
+
+const A_DAY_IN_SECS = int64(24 * 60 * 60)
+
+type LogPeriodicEntry_t struct {
+    ID      string          /* Identifier provided by caller */
+    Message string          /* string to log */
+    Lvl     syslog.Priority /* Log Level to use */
+    Period  int             /* period in seconds */
+    /* TODO: Change period to list of {period, cnt}
+     * after finishing cnt writes, move to next entry in list.
+     * Caller may send [ {5, 100}, {3600, 0} ], implying
+     * Write first 100 logs for every 5 seconds. After that
+     * write logs for every hour forever (cnt = 0) 
+     */
+}
+
+type LogPeriodicEntryInt_t struct {
+    LogPeriodicEntry_t
+    Due     int64           /* Next due epoch time point */
+}
+
+type LogPeriodic_t struct {
+    /* Channel to communicate to logging routine */
+    logCh               chan *LogPeriodicEntry_t
+
+    logPeriodicList     map[string]*LogPeriodicEntryInt_t
+    logPeriodicSorted   []*LogPeriodicEntryInt_t
+
+    /* TODO: Any entry after logging repeatedly at set period 
+     * for a day or two, reduce the period to every hour
+     * No point in polluting logs, as we have screamed enough
+     */
+}
+
+
+var logPeriodic *LogPeriodic_t
+
+func GetlogPeriodic() *LogPeriodic_t {
+    return logPeriodic
+}
+
+func LogPeriodicInit(chAbort chan interface{}) {
+    logPeriodic = &LogPeriodic_t {
+        logCh: make( chan *LogPeriodicEntry_t),
+        logPeriodicList: make(map[string]*LogPeriodicEntryInt_t),
+        logPeriodicSorted: nil,
+    }
+
+    go logPeriodic.run(chAbort)
+}
+
+func (p *LogPeriodic_t) AddLogPeriodic(l *LogPeriodicEntry_t) error {
+    if ((len(l.ID) == 0) || (len(l.Message) == 0)) {
+        return LogError("LogPeriodicEntry ID or message is empty")
+    }
+    min := GetConfigMgr().GetGlobalCfgInt("MIN_PERIODIC_LOG_PERIOD")
+    if l.Period < min {
+        return LogError("LogPeriodicEntry Period(%v) < min(%v)", l.Period, min)
+    }
+    p.logCh  <- l
+    return nil
+}
+
+func (p *LogPeriodic_t) DropLogPeriodic(ID string) {
+    if len(ID) > 0 {
+        /* Emtpy Message implies drop */
+        p.logCh  <- &LogPeriodicEntry_t {ID: ID }
+    }
+}
+
+
+func (p *LogPeriodic_t) run(chAbort chan interface{}) {
+    tout := A_DAY_IN_SECS           /* Just a init value; Once per day */
+
+    for {
+        upd := false
+        select {
+        case l := <- p.logCh:
+            upd = p.update(l)
+
+        case <- time.After(time.Duration(tout) * time.Second):
+            upd = p.WriteLogs()
+
+        case <- chAbort:
```

Comment:
Can we have a separate API which terminate periodic logging altogether ?

Reply:
LogPeriodicInit
Abort the last one and call this again to re-create.

Reply:
I mean just simple API without any channel. Let that API takes care of rest.